asim: print generated load, trace StoreRebalancer

tbg opened the pull request:
See individual commits.

Epic: CRDB-49117

tbg committed:
asim: print generated load in gen_load

And leave a couple of TODOs about stuff that is likely unintentional
or behavior that I couldn't understand.

Epic: CRDB-49117

tbg committed:
asim: rewrite high_cpu_25nodes.txt

The checked-in version seems to have come from a one-off run.

tbg committed:
asim: also trace StoreRebalancer

tbg committed:
asim: don't collect traces when we're not going to record them

This should speed up asim when not using `-rewrite`.

wenyihu6 reviewed (approved):
Sorry for the late review. Thanks for doing this! LGTM mod the comment about setting when enabled. I will push a fix and merge this. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154513#-:-OafoJSLBelaXAyDjDdx:b-di4w8q)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained

___
*[`pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/high_cpu_25nodes.txt` line 31 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/154513#-OafmYNw8q93m8Riw89b:-OafmYNw8q93m8Riw89c:beey56o) ([raw file](https://github.com/cockroachdb/cockroach/blob/5cc655917dfd8fa305dbc9bfff49a934049a8bd2/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/high_cpu_25nodes.txt#L31)):*
> ```help
> eval duration=20m samples=1 seed=42 cfgs=(mma-only,mma-count) metrics=(cpu,cpu_util,write_bytes_per_second,replicas,leases)
> ----
> cpu#1: last:  [s1=6305516202, s2=4203121714, s3=4209044827, s4=6316721595, s5=4197368968, s6=2107290360, s7=2099077812, s8=4207644575, s9=4199517611, s10=4195522966, s11=4205024256, s12=4198000753, s13=4202637211, s14=4186734543, s15=4199295119, s16=4199174047, s17=4204480499, s18=6290045651, s19=4193666150, s20=4211828495, s21=4201735135, s22=2096004946, s23=4201402387, s24=4193586357, s25=4193056496] (stddev=1029616922.87, mean=4200699947.00, sum=105017498675)
> ```

Hm, I don’t see any meaningful changes in this commit. Is the output from master wrong? Why is the nightly CI not failing. 
___
*[`pkg/kv/kvserver/asim/tests/datadriven_simulation_test.go` line 582 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/154513#-OafoD316QSDI7galAjd:-OafoD316QSDI7galAje:b-mdd7tg) ([raw file](https://github.com/cockroachdb/cockroach/blob/ea2242baef2d0f4b24a47b7516827054cf51f66a/pkg/kv/kvserver/asim/tests/datadriven_simulation_test.go#L582)):*
> ```Go
> 								if !tr.enabled {
> 									// Only populate OnRecording if we're going to save the results.
> 									// That way, we avoid creating trace spans during normal test runs.
> ```

Should we set it if tr.enabled is true? 


<!-- Sent from Reviewable.io -->

wenyihu6 committed:
asim: set settingsGen.Settings.OnRecording if tr is enabled

Previously, asim was setting settingsGen.Settings.OnRecording on
if tr is disabled which was unintentional. This commit fixes it
to set OnRecording when tr is enabled instead.

wenyihu6 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154513)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @tbg)

___
*[`pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/high_cpu_25nodes.txt` line 31 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/154513#-OafmYNw8q93m8Riw89b:-Oafr1kv9w75l2EncyG-:b-w31tyz) ([raw file](https://github.com/cockroachdb/cockroach/blob/5cc655917dfd8fa305dbc9bfff49a934049a8bd2/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/high_cpu_25nodes.txt#L31)):*
<details><summary><i>Previously, wenyihu6 (Wenyi Hu) wrote…</i></summary><blockquote>

Hm, I don’t see any meaningful changes in this commit. Is the output from master wrong? Why is the nightly CI not failing.
</blockquote></details>

I re-ran this test with the rewrite and saw that the previous version of this file was back. I’ll revert this commit before bors. Hope it wasn't because the test is flaky.


<!-- Sent from Reviewable.io -->

wenyihu6 committed:
Revert "asim: rewrite high_cpu_25nodes.txt"

This reverts commit 5cc655917dfd8fa305dbc9bfff49a934049a8bd2.
